opennebula/resource_opennebula_virtual_machine.go

```diff
@@ -540,16 +540,21 @@ func resourceOpennebulaVirtualMachineReadCustom(ctx context.Context, d *schema.R
 		})
 		return diags
 	}
-	d.SetId(fmt.Sprintf("%v", vm.ID))
-	d.Set("name", vm.Name)
-	d.Set("uid", vm.UID)
-	d.Set("gid", vm.GID)
-	d.Set("uname", vm.UName)
-	d.Set("gname", vm.GName)
-	d.Set("state", vm.StateRaw)
-	d.Set("lcmstate", vm.LCMStateRaw)
+	d.SetId(fmt.Sprintf("%v", vmInfo.ID))
+	d.Set("name", vmInfo.Name)
+	d.Set("uid", vmInfo.UID)
+	d.Set("gid", vmInfo.GID)
+	d.Set("uname", vmInfo.UName)
+	d.Set("gname", vmInfo.GName)
+	d.Set("state", vmInfo.StateRaw)
+	d.Set("lcmstate", vmInfo.LCMStateRaw)
+	if vm.State(vmInfo.StateRaw) == vm.Done {
```

Comment:
Is it the proper way to do ? 
There's a `-replace` option to terraform
And if we consider it's ok, do we want this behavior everytime, for everyone or should it be conditioned by an other attribute with a name ?

Reply:
As far as I know, unsetting the "ID" parameter is the way to tell terraform that the previously existing resource no longer exists (see an example on AWS's provider [here](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/ec2_instance.go#L1125).

A VM in `done` state can be considered as non-existing anymore. As it is expected that terraform recreates the resource when they do not exist anymore, this behaviour should be default.

An example for that expectation is drift checks, if you create a VM, someone deletes it from the UI, the expectation is that terraform will tell you there is a drift.

The thing to consider is the non-backward compatible behaviour this brings to the vm resource, this might be worth of a major version release.

Maybe this could be behind a `recreate_on_delete` boolean on the resource on an initial minor update, with a deprecation notice for the next major update.
This also affects the `vrouter` resource.

Reply:
@frousselet what do you think on this ?
Should we consider this behavior to be protected by a `recreate_on_delete` attribute ?